scripts/setup-rg/README.md

```diff
@@ -0,0 +1,66 @@
+# Resource Group Setup
+
+These scripts are helpers for setting up a new resource group (RG) as the control plane for MLOS.
+
+## Quickstart
+
+1. Starting in this current directory, ensure that we are logged in to Azure CLI.
+
+    ```sh
+    az login
+    ```
+
+2. Make a copy of the ARM parameters file.
+
+    ```sh
+    cp rg-template.example.parameters.json rg-template.<your name>.parameters.json
+    ```
+
+3. Modify the ARM parameters in the newly created file as needed, especially the `PLACEHOLDER` values.
+
+4. Execute the main script and follow prompts:
+
+    ```sh
+    ./setup-rg.ps1
+    Supply values for the following parameters:
+    armParameters: rg-template.<your name>.parameters.json
+    servicePrincipalName: <name of service principal for generating tokens with>
+    resourceGroupName: <target resource group for this control plane setup>
+    certName: <desired key vault name of certificate for the service principal e.g. mlos-autotune-sp-cert>
+    ```
+
+## Manual
+
+Parameters for script can also passed in manually (without prompt) as follows:
+
+```sh
+./setup-rg.ps1 `
+    -armParameters $armParams `
+    -servicePrincipalName $servicePrincipalName `
+    -resourceGroupName $resourceGroupName `
+    -certName $certName
+```
+
+where `$armParams` follows the same usage as `--parameters` in [az deployment group create](https://learn.microsoft.com/en-us/cli/azure/deployment/group?view=azure-cli-latest#az-deployment-group-create-examples).
+
+## Workflow
+
+The high-level flow for what this script automates is as follows:
+
+1. Assign `Contributor` access to the Service Principal (SP) for write access over resources.
+    For now we assume the experiment's resources are provisioned in the same RG as the control plane, so the access is granted to the same RG.
+
+2. Provision control plane resources into the RG.
+    This includes:
+    - Networking (public IP, security group, vnet, subnet, network interface)
+    - Key Vault
+    - Storage (storage account, file share, MySQL Flex server)
```

Comment:
```suggestion
    - Storage (storage account, file share, DB)
```

- DB need not be MySQL, so we should make that optional as well (can certainly default to MySQL though and let it get parameterized).
- Storage needs to be location dependent, and embedded in the name.
  As such, we should make sure this script can be idempotent or split so that we can independently setup storage in multiple locations without it complaining about the DB portion or scheduler VM already existing.

In fact as far as that goes, we could even have multiple scheduler VMs at some point.

Reply:
Nm, in the templates that gets hard to support.

We can stick with a MySQL Flex DB for now, but let's do the following:

1. Separate out the template so that some of these resources can be deployed, in an appropriate order, separately.
2. Update the script to allow skip deploying the DB in case someone wants to provide an alternative implementation option there.  Our default supported implementation can be MySQL for now.
3. Label it generically as "results DB" instead.

Reply:
- Split out an ARM template for the results DB that users can optionally provision.
- Kept storage in the original template, but now allows for users to set a list of locations to create one storage for each, with location is appended to the storage name. This was ease the updating of the control vnet's service endpoints too.

Reply:
On the last point, will that still work if we want to add additional locations after the fact, or do they all needed to be added at the initial creation time?

Reply:
It will still work after the fact, it will incrementally add the storages!